sql-db/panache-flyway/src/test/java/io/quarkus/ts/sqldb/panacheflyway/dbpool/AgroalPoolTest.java

```diff
@@ -137,23 +149,36 @@ public void borderConditionBetweenIdleAndGetConnectionTest() {
             subscriber
                     .awaitItems(CONCURRENCY_LEVEL)
                     .getItems()
-                    .forEach(statusCode -> assertEquals(statusCode, HttpStatus.SC_OK, "Unexpected Application response"));
+                    .forEach(statusCode -> assertEquals(HttpStatus.SC_OK, statusCode,
+                            "Unexpected response status: " + statusCode));
+
+            subscriber.cancel();
         }
     }
 
+    @Order(4)
     @Test
-    public void concurrentLoadTest() {
+    public void concurrentLoadTest() throws InterruptedException {
         final int events = 100;
+        final CountDownLatch latch = new CountDownLatch(events * CONCURRENCY_LEVEL);
         for (int i = 0; i < events; i++) {
             Multi.createFrom()
                     .range(0, CONCURRENCY_LEVEL).subscribe()
-                    .with(n -> assertEquals(2, users.count(), "UnexpectedUser Amount"));
+                    .with(n -> {
+                        assertEquals(2, users.count(), "UnexpectedUser Amount");
+                        latch.countDown();
+                    });
         }
+        var allUserCountsDone = latch.await(2, TimeUnit.SECONDS);
```

Comment:
I would prefer to have a boolean here, instead of var. It's IMHO easier to read.

Reply:
Yes, you are right. And even more, IIRC if you assign boolean to `var` it becomes boxed. I think I just typed it by habit. I'll fix it.